snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -453,13 +453,20 @@ def _get_fileset(self, option, default=None):
     def _organize(self):
         fileset = self._get_fileset('organize', {})
 
-        _organize_filesets(fileset.copy(), self.code.installdir)
+        _organize_filesets(fileset.copy(), self.stagedir)
 
     def stage(self, force=False):
         self.makedirs()
         self.notify_part_progress('Staging')
-        self._organize()
         snap_files, snap_dirs = self.migratable_fileset_for('stage')
+        organize_fileset = getattr(self.code.options, 'organize', {}) or {}
+
+        snap_files, new_snap_dirs = _organize_fileset(
+            snap_files, organize_fileset, self.code.installdir)
+        snap_dirs, extra_snap_dirs = _organize_fileset(
+            snap_dirs, organize_fileset, self.code.installdir)
+
+        snap_dirs |= new_snap_dirs | extra_snap_dirs
```

Comment:
This chunk of code looks an _awful_ lot like `_get_staged_files()`, below. Should it be using that?